module/PowerNSX.psm1

```diff
@@ -13916,6 +13993,51 @@ function Set-NsxEdge {
             $_Edge.RemoveChild($edgeSummary) | out-null
         }
 
+        #cliSettings
+        if ( $PsBoundParameters.ContainsKey('userName') ) {
+            if ( $PsBoundParameters.ContainsKey('password') ) {
+                if ( invoke-xpathquery -node $_Edge -querymethod SelectSingleNode -Query "child::cliSettings/userName" ) {
+                    $_Edge.cliSettings.username = $userName
+                } else {
+                    Add-XmlElement -xmlroot $_Edge.cliSettings -xmlElementName "userName" -xmlElementText $userName
+                }
+            } else {
+                throw "You need to specify a password for change username..."
+            }
+        }
+
+        if ( $PsBoundParameters.ContainsKey('password') ) {
+            if ( invoke-xpathquery -node $_Edge -querymethod SelectSingleNode -Query "child::cliSettings/password" ) {
+                $_Edge.cliSettings.password = $password
+            } else {
+                Add-XmlElement -xmlRoot $_Edge.cliSettings -xmlElementName "password" -xmlElementText $password
+            }
+        }
+
+        if ( $PsBoundParameters.ContainsKey('remoteAccess') ) {
+            if ( invoke-xpathquery -node $_Edge -querymethod SelectSingleNode -Query "child::cliSettings/remoteAccess" ) {
+                $_Edge.cliSettings.remoteAccess = $remoteAccess.ToString().ToLower()
+            } else {
+                Add-XmlElement -xmlroot $_Edge.cliSettings -xmlElementName "remoteAccess" -xmlElementText $remoteAccess.ToString().ToLower()
+            }
+        }
+
+        if ( $PsBoundParameters.ContainsKey('passwordExpiry') ) {
+            if ( invoke-xpathquery -node $_Edge -querymethod SelectSingleNode -Query "child::cliSettings/passwordExpiry" ) {
+                $_Edge.cliSettings.passwordExpiry = $passwordExpiry
+            } else {
+                Add-XmlElement -xmlroot $_Edge.cliSettings -xmlElementName "passwordExpiry" -xmlElementText $passwordExpiry
```

Comment:
This line fails the windows tests
```
      [-] Change Password Expiry 277ms
        Cannot set "passwordExpiry" because only strings can be used as values to set XmlNode properties.
        at <ScriptBlock>, C:\jenkins\workspace\WindowsPowerNsx636\tests\integration\04.Edge.Tests.ps1: line 762
        762:             Get-NsxEdge $name | Set-NsxEdge -passwordExpiry 4242 -confirm:$false
```
Change to `$passwordExpiry.ToString()` and it should pass the windows tests.